clang-toolchain/Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Copyright 2019 Tetrate
```

Comment:
should we add a README explaining what this is, how to run it (ie run it in your own cloud build, I suppose)?

Reply:
We can do that, but I think it would be even better to just set-up the trigger in `istio-testing`. I didn't have GitHub permissions to do it.